manager/apis/app/v1alpha1/blueprint_types.go

```diff
@@ -5,87 +5,60 @@ package v1alpha1
-type ReadModuleArgs struct {
-	// Source of the read path module
-	// +required
-	Source DataStore `json:"source"`
+	// Destination is the data store to which the data will be written
+	// +optional
+	Destination *DataStore `json:"destination,omitempty"`
 
 	// AssetID identifies the asset to be used for accessing the data when it is ready
 	// It is copied from the FybrikApplication resource
 	// +required
 	AssetID string `json:"assetID"`
 
-	// Transformations are different types of processing that may be done to the data
+	// Transformations are different types of processing that may be done to the data as it is copied.
 	// +optional
 	Transformations []taxonomy.Action `json:"transformations,omitempty"`
 }
 
-// WriteModuleArgs define the input parameters for modules that write data to location B
-type WriteModuleArgs struct {
-	// Destination is the data store to which the data will be written
-	// +required
-	Destination DataStore `json:"destination"`
-
-	// AssetID identifies the asset to be used for accessing the data when it is ready
-	// It is copied from the FybrikApplication resource
+type ApplicationDetails struct {
+	// UUID of FybrikApplication
 	// +required
-	AssetID string `json:"assetID"`
-
-	// Transformations are different types of processing that may be done to the data as it is written.
-	// +optional
-	Transformations []taxonomy.Action `json:"transformations,omitempty"`
-}
+	UUID string `json:"uuid"`
 
-// ModuleArguments are the parameters passed to a component that runs in the data path
-// In the future might support output args as well
-// The arguments passed depend on the type of module
-type ModuleArguments struct {
 	// Labels of FybrikApplication
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
 
 	// Application selector is used to identify the user workload.
 	// It is obtained from FybrikApplication spec.
 	// +optional
-	AppSelector metav1.LabelSelector `json:"appSelector,omitempty"`
+	WorkloadSelector metav1.LabelSelector `json:"selector,omitempty"`
 
-	// CopyArgs are parameters specific to modules that copy data from one data store to another.
+	// Application context such as intent, role, etc.
 	// +optional
-	Copy *CopyModuleArgs `json:"copy,omitempty"`
-
-	// ReadArgs are parameters that are specific to modules that enable an application to read data
-	// +optional
-	Read []ReadModuleArgs `json:"read,omitempty"`
+	Context taxonomy.AppInfo `json:"context,omitempty"`
+}
 
-	// WriteArgs are parameters that are specific to modules that enable an application to write data
+// ModuleArguments are the parameters passed to a component that runs in the data path
+type ModuleArguments struct {
+	// Assets define asset related arguments, such as data source, transformations, etc.
 	// +optional
-	Write []WriteModuleArgs `json:"write,omitempty"`
+	Assets []AssetContext `json:"assets,omitempty"`
+	// Logging verbosity
+	// +required
+	Verbosity zerolog.Level `json:"verbosity"`
```

Comment:
Do we need to pass the verbosity as a parameter?  The modules take it from a config file.

Reply:
A config file is for manager. It is deployed in fybrik-system. Modules will run in a different namespace. They can not access the config map in fybrik-system.

Reply:
we need to add `prettyLogging` flag as well. Maybe we can combine all logging configuration fields in a sub-struct. 
Should Verbosity be a required parameter, usually most of the systems have a defult log level.

Reply:
My thinking is that we do not need to pass the logging information to the modules.  They are independent components that can set their logging levels as they see fit.  If you feel otherwise though, I guess there is no harm in passing them.  It's just that currently the arrow-flight-module is being updated to get the info from a config file.

Reply:
agree, especially that modules can be implemented in different languages, with different log levels, logging config parameters.

Reply:
removed logging.